libr/include/r_anal.h

```diff
@@ -1315,6 +1315,7 @@ R_API bool r_anal_bb_set_offset(RAnalBlock *bb, int i, ut16 v);
 R_API ut16 r_anal_bb_offset_inst(RAnalBlock *bb, int i);
 R_API ut64 r_anal_bb_opaddr_at(RAnalBlock *bb, ut64 addr);
 R_API bool r_anal_bb_op_starts_at(RAnalBlock *bb, ut64 addr);
+R_API RAnalBlock *r_anal_bb_op_starts_at_which(RAnalBlock **bbs, ut64 addr);
```

Comment:
I think this API is not as good as the rest. It accepts a `RAnalBlock**`.. this array comes from `bbget_all` and it's only used in that case. I see no reasons to have bbget_all and r_anal_bb_op_starts_at_which.

I think you can just merge the two things in something like:
```
R_API RAnalBlock *r_anal_bb_op_starts_at_which(RAnalFunction *fcn, ut64 addr) {
    r_list_foreach (fcn->bbs, iter, bb) {
		ut64 eaddr = bb->addr + bb->size;
		if ((bb->addr >= eaddr && addr == bb->addr) || (addr >= bb->addr && addr < eaddr)) {
			if (r_anal_bb_op_starts_at (bb, addr)) {
			         return bb;
		        }
		}
	}
}
```

And:
```
bbs = bbget_all (fcn, addr + idx);
bbg = bbs ? r_anal_bb_op_starts_at_which (bbs, addr + idx) : NULL;
free (bbs);
```
would become:
```
bbg = r_anal_op_stars_at_which (fcn, addr + idx);
```

(ofcourse this is not tested and it doesn't compile, but you get the idea)

Reply:
I know ... both bbget_all and r_anal_bb_op_starts_at_which are slated to be removed in a subsequent PR but the code now currently works and I want to keep this code for now so that I have something to compare against later.

Reply:
Just do a comparison in local between the two versions. This API as it is is almost useless now, because it's just used in that place and it won't be used anywhere else, because it requires RAnalBlock**. Otherwise, make it static, but IMHO it should just be done well from the beginning. It's not such a big change.

We all know the "we'll fix it later" often becomes a "it will remain that way for at least few months/years".

Reply:
ok 2814f3492cb34109931e787b86a3235d27520ce3

> We all know the "we'll fix it later" often becomes a "it will remain that way for at least few months/years".

Actually, this one has higher priority than most because ... ok I was just being silly

Reply:
Thanks a lot, I appreciate :) It's much better now.